### PR TITLE
Part 2 #69 Switch UI and data scripts to duration field in narration model

### PR DESCRIPTION
**Pull request by frombrest**
Hi @nbeloglazov, I have prepared changes to start using duration field from narration model. I hope I found all places which must be fixed. Also, could you help please with testing especially data scripts. 
As far as I understand autotests are failing due to `duration_sec` field of book model in data.json (test data set). Do I need to delete it from [belaudiobooks/data](https://github.com/belaudiobooks/data) repo?

**Attention!:** This PR need to be merged only after duration data will be completely copied/moved to narration table since 0004 migration delete this column from book table:
- [ ] ok

### CODE REVIEW

File: templates/books/articles/audiobooksby-database.html [hunk 1]
```diff
@@ -29,7 +29,12 @@ <h2>Навошта data.json?</h2>
   <pre>
 // run from devtools on data.json page
 const data = JSON.parse(document.body.innerText);
-data.books.reduce((a, b) => a.duration_sec > b.duration_sec ? a : b)
+data.books.reduce((a, b) =>
+    a.narrations.reduce(
+      (c, d) => c.duration > d.duration ? c : d
+    ).duration > b.narrations.reduce(
+      (e, f) => e.duration > f.duration ? e : f
+    ).duration ? a : b)
```
Comment (nbeloglazov) [inline templates/books/articles/audiobooksby-database.html:37]: This script seems to have few errors. Also it becomes quite big so not the best candidate as example. Let's replace it with a new script: "Знайсці аўдыякнігу з найбольшай колькасцю агучак":

```
const data = JSON.parse(document.body.innerText);
data.books.reduce((a, b) => a.narrations.length > b.narrations.length ? a : b);
```

File: templates/books/articles/audiobooksby-database.html [hunk 2]
```diff
@@ -75,6 +79,7 @@ <h2>Схема data.json</h2>
   // list of Person.uuid
   narrators: string[];
   links: Links[];
+  duration: string;
```
Comment (nbeloglazov) [inline templates/books/articles/audiobooksby-database.html:82]: Is it string or number? The old duration_sec was number.

**Review by nbeloglazov — CHANGES_REQUESTED** [review]

File: templates/books/book-detail.html [hunk 3]
```diff
@@ -50,11 +50,6 @@ <h1 class="h2">{{ book.title }}</h1>
                 <div class="mb-2">{{ book.title_ru }}</div>
             {% endif %}
 
-            <!--Book Duration-->
-            {% if book.duration_sec %}
-                <p>{{ book | duration }}</p>
-            {% endif %}
```
Comment (nbeloglazov) [inline templates/books/book-detail.html:56]: Depending on whether book has a single or multiple narrations we use slightlhy different layouts. For example to render narrator:

* with single narration we render narrator next to author (book level): https://audiobooks.by/books/rukh

* with multiple narrations we render narrators at the narration level: https://audiobooks.by/books/liudzi-na-balotse

We should do the same with duration. The rationale is that displaying duration at book level visually better and majority of the books 95+% have only single narration. So we should optimize for that case.

**Review by nbeloglazov — CHANGES_REQUESTED** [review]

File: templates/books/articles/audiobooksby-database.html (hunk 2 above)
Comment (frombrest) [reply templates/books/articles/audiobooksby-database.html:82]: Unless I missed custom serialization logic somewhere...  Django's `DurationField` must be serialized to string:

venv/lib/python3.9/site-packages/django/utils/duration.py
```
def duration_string(duration):
    """Version of str(timedelta) which is not English specific."""
    days, hours, minutes, seconds, microseconds = _get_duration_components(duration)

    string = '{:02d}:{:02d}:{:02d}'.format(hours, minutes, seconds)
    if days:
        string = '{} '.format(days) + string
    if microseconds:
        string += '.{:06d}'.format(microseconds)

    return string
```

But I may be wrong.

Comment (frombrest) [reply templates/books/articles/audiobooksby-database.html:82]: Yeah, make sense to keep it as number of seconds. Let me fix that.

File: templates/books/book-detail.html (hunk 3 above)
Comment (frombrest) [reply templates/books/book-detail.html:56]: I changed a bit template, so please check again. Is that what you mean?

File: templates/books/book-detail.html [hunk 4]
```diff
@@ -98,6 +100,9 @@ <h1 class="h2">{{ book.title }}</h1>
                             Агучка: удакладняецца...
                         {% endif %}
                         <br>
+                        {% if narration.duration %}
```
Comment (nbeloglazov) [inline templates/books/book-detail.html:103]: We should render this only when there are two or more narrations. So need to add an if similar to above.

**Review by nbeloglazov — CHANGES_REQUESTED** [review]

File: templates/books/book-detail.html (hunk 3 above)
Comment (nbeloglazov) [reply templates/books/book-detail.html:56]: Yes, but left comment below.

File: templates/books/book-detail.html (hunk 4 above)
Comment (nbeloglazov) [reply templates/books/book-detail.html:103]: Ah, nvm. Didn't notice that this is already in that "> narrations" section.

**Pull request merged**